Microsoft.Azure.CosmosRepository/src/InMemoryRepository.cs

```diff
@@ -105,23 +119,49 @@ public async ValueTask<IEnumerable<TItem>> CreateAsync(IEnumerable<TItem> values
         }
 
         /// <inheritdoc/>
-        public async ValueTask<TItem> UpdateAsync(TItem value, CancellationToken cancellationToken = default)
+        public async ValueTask<TItem> UpdateAsync(TItem value, CancellationToken cancellationToken = default, bool verifyEtag = false)
         {
             await Task.CompletedTask;
 
-            Items[value.Id] = SerializeItem(value);
+            if (Items.ContainsKey(value.Id))
+            {
+                if (verifyEtag)
+                {
+                    if (value is IItemWithEtag valueWithEtag)
+                    {
+                        TItem existingItem = DeserializeItem(Items[value.Id]);
+                        if (existingItem is IItemWithEtag existingItemWithEtag)
+                        {
+                            if (existingItemWithEtag.Etag != valueWithEtag.Etag)
+                            {
+                                MismatchedEtags();
+                            }
+                        }
+                        else
+                        {
+                            throw new InvalidEtagConfigurationException(string.Empty);
+                        }
+                    }
+                    else
+                    {
+                        throw new InvalidEtagConfigurationException(string.Empty);
+                    }
+                }
+            }
```

Comment:
```suggestion
            if (Items.ContainsKey(value.Id))
            {
                if (verifyEtag && value is IItemWithEtag valueWithEtag)
                {
                    TItem existingItem = DeserializeItem(Items[value.Id]);
                    if (existingItem is IItemWithEtag existingItemWithEtag)
                    {
                        if (existingItemWithEtag.Etag != valueWithEtag.Etag)
                        {
                            MismatchedEtags();
                        }
                    }
                    else
                    {
                        throw new InvalidEtagConfigurationException(string.Empty);
                    }
                }
                else
                {
                    throw new InvalidEtagConfigurationException(string.Empty);
                }
            }
```

So many nested `if` 😄 . We should work on streaming this a bit.